### PR TITLE
added simplemd and compiled md body to html on server

### PR DESCRIPTION
**Pull request by yuchunc**
這個先不要 Merge, Guardian 目前有 [Bug](https://github.com/ueberauth/guardian/issues/242)。

### CODE REVIEW

File: web/static/vendor/simplemde.min.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
...
...w-side>p,.editor-preview>p{margin-top:0}.editor-preview pre,.editor-preview-side pre{background:#eee;margin-bottom:10px}.editor-preview table td,.editor-preview table th,.editor-preview-side table td,.editor-preview-side table th{border:1px solid #ddd;padding:5px}.CodeMirror .CodeMirror-code .cm-tag{color:#63a35c}.CodeMirror .CodeMirror-code .cm-attribute{color:#795da3}.CodeMirror .CodeMirror-code .cm-string{color:#183691}.CodeMirror .CodeMirror-selected{background:#d9d9d9}.CodeMirror .CodeMirror-code .cm-header-1{font-size:200%;line-height:200%}.CodeMirror .CodeMirror-code .cm-header-2{font-size:160%;line-height:160%}.CodeMirror .CodeMirror-code .cm-header-3{font-size:125%;line-height:125%}.CodeMirror .CodeMirror-code .cm-header-4{font-size:110%;line-height:110%}.CodeMirror .CodeMirror-code .cm-comment{background:rgba(0,0,0,.05);border-radius:2px}.CodeMirror .CodeMirror-code .cm-link{color:#7f8c8d}.CodeMirror .CodeMirror-code .cm-url{color:#aab2b3}.CodeMirror .CodeMirror-code .cm-strikethrough{text-decoration:line-through}.CodeMirror .CodeMirror-placeholder{opacity:.5}.CodeMirror .cm-spell-error:not(.cm-url):not(.cm-comment):not(.cm-tag):not(.cm-word){background:rgba(255,0,0,.15)}
```
Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

Comment (yuchunc) [inline web/static/vendor/simplemde.min.css]: Colon after property should be followed by one space

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spaceafterpropertycolon)

**Comment by yuchunc** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 935 possible new issues (including those that may have been commented here).
* 1 fixed issue! 🎉

You can see more details about this review at https://ebertapp.io/github/elixirtw/elixir_tw/pulls/24.

**Comment by otaq** [comment]
brunch 這樣設定就不用在 vendor 放 simplemde 的程式碼

```js
// brunch-config.js
      jQuery: 'jquery',
      SimpleMDE: 'simplemde'
    },
    static: ['node_modules/semantic-ui-css/semantic.min.js'],
    styles: {
      'semantic-ui-css': ['semantic.css'],
      'simplemde': ['dist/simplemde.min.css']
    },
```

**Comment by yuchunc** [comment]
@otaq 酷！我來加
謝謝

File: web/static/css/layout.scss [hunk 2]
```diff
@@ -11,3 +11,9 @@
   flex: 1;
 }
 
+#main-nav {
+  a {
+    color: white;
```
Comment (yuchunc) [inline web/static/css/layout.scss]: Color `white` should be written in hexadecimal form as `#ffffff`

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#colorkeyword)

Comment (yuchunc) [inline web/static/css/layout.scss]: Properties should be ordered font-weight, color

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

File: web/static/css/layout.scss [hunk 3]
```diff
@@ -11,3 +11,9 @@
   flex: 1;
 }
 
+#main-nav {
```
Comment (yuchunc) [inline web/static/css/layout.scss]: Avoid using id selectors

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#idselector)

File: web/controllers/post_controller.ex [hunk 4]
```diff
@@ -5,20 +5,21 @@ defmodule ElixirTw.PostController do
 
   def index(conn, _params) do
     posts = fetch_posts
-
+            |> Enum.map( &update_comments_count/1 )
```
Comment (yuchunc) [inline web/controllers/post_controller.ex]: There is no whitespace around parentheses/brackets most of the time, but here there is.

**Comment by yuchunc** [comment]
Works! 前端套件管理果然是個神奇的世界 @@

**Comment by yuchunc** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 4 possible new issues (including those that may have been commented here).
* 1 fixed issue! 🎉

But beware that this branch is **[2 commits behind the `elixirtw:master` branch](https://github.com/elixirtw/elixir_tw/compare/yuchunc:public_post_view...elixirtw:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/elixirtw/elixir_tw/pulls/24.

**Pull request merged**